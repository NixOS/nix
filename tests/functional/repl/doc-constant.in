:l doc-comments.nix
:doc constant
:doc lib.version
:doc lib.attr.empty
:doc lib.attr.undocument
:doc (import ./doc-comments.nix).constant
:doc (import ./doc-comments.nix).lib.version
:doc (import ./doc-comments.nix).lib.attr.empty
:doc (import ./doc-comments.nix).lib.attr.undocumented
:doc missing
:doc constanz
:doc missing.attr
:doc lib.missing
:doc lib.missing.attr
:doc lib.attr.undocumental
