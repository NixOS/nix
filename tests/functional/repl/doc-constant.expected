Nix <nix version>
Type :? for help.

nix-repl> :l doc-comments.nix
Added <number omitted> variables.

nix-repl> :doc constant
error: value does not have documentation

nix-repl> :doc lib.version
Attribute `version`

  … defined at /path/to/tests/functional/repl/doc-comments.nix:30:3

Immovably fixed.

nix-repl> :doc lib.attr.empty
Attribute `empty`

  … defined at /path/to/tests/functional/repl/doc-comments.nix:33:3

Unchangeably constant.

nix-repl> :doc lib.attr.undocument
error:
       … while evaluating the attribute 'attr.undocument'
         at /path/to/tests/functional/repl/doc-comments.nix:33:3:
           32|   /** Unchangeably constant. */
           33|   lib.attr.empty = { };
             |   ^
           34|

       error: attribute 'undocument' missing
       at «string»:1:1:
            1| lib.attr.undocument
             | ^
       Did you mean undocumented?

nix-repl> :doc (import ./doc-comments.nix).constant
Attribute `constant`

  … defined at /path/to/tests/functional/repl/doc-comments.nix:27:3

Firmly rigid.

nix-repl> :doc (import ./doc-comments.nix).lib.version
Attribute `version`

  … defined at /path/to/tests/functional/repl/doc-comments.nix:30:3

Immovably fixed.

nix-repl> :doc (import ./doc-comments.nix).lib.attr.empty
Attribute `empty`

  … defined at /path/to/tests/functional/repl/doc-comments.nix:33:3

Unchangeably constant.

nix-repl> :doc (import ./doc-comments.nix).lib.attr.undocumented
Attribute `undocumented`

  … defined at /path/to/tests/functional/repl/doc-comments.nix:35:3

No documentation found.

nix-repl> :doc missing
error: undefined variable 'missing'
       at «string»:1:1:
            1| missing
             | ^

nix-repl> :doc constanz
error: undefined variable 'constanz'
       at «string»:1:1:
            1| constanz
             | ^

nix-repl> :doc missing.attr
error: undefined variable 'missing'
       at «string»:1:1:
            1| missing.attr
             | ^

nix-repl> :doc lib.missing
error: attribute 'missing' missing
       at «string»:1:1:
            1| lib.missing
             | ^

nix-repl> :doc lib.missing.attr
error: attribute 'missing' missing
       at «string»:1:1:
            1| lib.missing.attr
             | ^

nix-repl> :doc lib.attr.undocumental
error:
       … while evaluating the attribute 'attr.undocumental'
         at /path/to/tests/functional/repl/doc-comments.nix:33:3:
           32|   /** Unchangeably constant. */
           33|   lib.attr.empty = { };
             |   ^
           34|

       error: attribute 'undocumental' missing
       at «string»:1:1:
            1| lib.attr.undocumental
             | ^
       Did you mean undocumented?
